core/src/main/java/com/nisovin/magicspells/volatilecode/VolatileCodeHandle.java

```diff
@@ -52,4 +54,8 @@
 	ItemStack setNBTString(ItemStack item, String key, String value);
 
 	String getNBTString(ItemStack item, String key);
+
+	boolean addCookingRecipe(String type, NamespacedKey namespace, String group, ItemStack result, Material ingredient, float experience, int cookingTime);
+
+	boolean addStonecutterRecipe(NamespacedKey namespace, String group, ItemStack result, Material ingredient);
```

Comment:
Are you unable to just return the recipe object? Recipe is a type that exists across all supported versions and is a a common supertype for all of them. Registry of the recipes does not belong in here as far as I can see.